Upload view: Display upload limits to the user in upload view

ThomasFinnern opened the pull request:
Intention: Tell the user which upload limit is active in the upload form where he may need it.

The user sees one line with a calculated maximum upload limit. It is calculated from the minum of the other variables shown when opening the aditional lines.
The limits are defined in php.ini and joomgalley config. See description behind the values.

![Limits by php.ini and config](https://github.com/ThomasFinnern/JoomGallery_fith_dev/blob/main/.jg_dev_doc/jg_4x/images/site.UserUpload/userUpload.maxlimit.en.png?raw=true  "")

How to test: 

1) It should display, and open all lines on click 
2) Check values in php.ini and config for matching 
3) Change values there when two lines have the same to be sure it adresses the right vaiable

MrMusic commented:
New language constants are missing in the language file.

Backend upload limit from joomla media manager is not taken into account.
JoomGallery config limit is only for frontend uploads.
See screenshots:
![upload_limit_media_manager](https://github.com/user-attachments/assets/0957b5a3-748c-4e3f-b0db-5bd1df2bf38e)
![upload_limit_frontend](https://github.com/user-attachments/assets/44917a1b-8a83-4799-8d44-8348ce8619d1)

ThomasFinnern commented:
Fixed: "New language constants are missing in the language file."
Fixed: Backend upload limit from joomla media manager is not taken into account.

if com_media Maximum Size (in MB) = 0 ->unlimited:  only the php limits are checked

@MrMusic thanks, please test again

ThomasFinnern commented on administrator/com_joomgallery/language/en-GB/com_joomgallery.ini:
```diff
@@ -926,5 +926,11 @@ COM_JOOMGALLERY_MENU_CATEGORY_VIEW_PAGINATION_OPT2="Load more button"
 COM_JOOMGALLERY_MENU_CATEGORY_VIEW_NUMBER_OF_RELOADED_IMAGES_DESC="Number of reloaded images per load more click."
 COM_JOOMGALLERY_MENU_CATEGORY_VIEW_NUMBER_OF_RELOADED_IMAGES_LABEL="# Images Reloaded"
 
+COM_JOOMGALLERY_UPLOAD_UPLOAD_LIMIT_IS="Maximum upload size: "
+COM_JOOMGALLERY_UPLOAD_POST_MAX_SIZE_IS="Maximum post size: "
+COM_JOOMGALLERY_UPLOAD_POST_MEMORY_LIMIT_IS="Maximum script limit: "
+COM_JOOMGALLERY_UPLOAD_MEDIA_LIMIT_IS="J! media upload limit: "
+COM_JOOMGALLERY_POST_MAX_SIZE_IS="Max upload limit (calculated bytes):&nbsp;<strong>%s</strong>&nbsp;MB"
+
```
I will improve parts. The string in brackets was separated for better alignment in view

ThomasFinnern commented:
2025.07.09 
Reorganized the layout for three column for better alingment of value and MB ... text
See result: 
![MaxUploadLimitBackend](https://github.com/user-attachments/assets/e70c385b-db34-4acf-81c8-710400313238)

Do we really need to translate PHP / MB / ... ?

Elfangor93 commented:
Please rebase this PR, such that is contains the commits of v4.1.0. Then the checks will succeed and the PR can be properly installed for testing. Thanks!

MrMusic commented:
Thanks for your changes and your intentions.
My intention was to move the text that comes after 'MB' to the left side to make a better look.
The 'PHP' in the text should make clear that some values are given by the PHP settings.
But we can also leave it as it is now. Thank you.

Elfangor93 commented:
@ThomasFinnern Performed the rebase with v4.2.0 branch

ThomasFinnern commented:
Will do and finish this week

MrMusic commented on administrator/com_joomgallery/tmpl/image/upload.php:
```diff
@@ -123,3 +126,95 @@
+                    <strong><?php echo $UploadLimit; ?></strong>
+                  </td>
+                  <td class="d-md-table-cell ps-1  text-start">
+                    MB (PHP 'upload_max_filesize')"
```
The `"` at the end is superfluous

ThomasFinnern commented:
Fixed

MrMusic commented:
I have tested this item ✅ successfully.
Thank you very much!
